contracts/MarginBase.sol

```diff
@@ -686,15 +679,15 @@ contract MarginBase is MinimalProxyable, IMarginBase, OpsReady {
         return IFuturesMarket(futuresManager.marketForKey(_marketKey));
     }
 
-    /// @notice exchangeRates() fetches current ExchangeRates contract
-    function exchangeRates() internal view returns (IExchangeRates) {
-        return
-            IExchangeRates(
-                addressResolver.requireAndGetAddress(
-                    "ExchangeRates",
-                    "MarginBase: Could not get ExchangeRates"
-                )
-            );
+    /// @notice get exchange rate of underlying market asset in terms of sUSD
+    /// @param market: synthetix futures market
+    /// @return price in sUSD
+    function sUSDRate(IFuturesMarket market) internal view returns (uint256) {
+        (uint256 price, bool invalid) = market.assetPrice();
+        if (invalid) {
+            revert InvalidPrice();
+        }
```

Comment:
can you also test this case? seems like critical functionaliy

Reply:
Would need to be done I think via mocking, right? Replicating this is forked environment would (again, I think) be difficult. Can mock easily with foundry

Reply:
yes mocking, just wanted to make sure this flow is covered

Reply:
Added!